content/blog/eng/from-0-to-infinite-the-final-keyword.md

```diff
@@ -0,0 +1,338 @@
+
+
+Obviously it is not always all pink and flowers, not always literature especially in our work, it is perfectly suited to all needs. 
+What I want to point out is that in some cases **final** can be a good ally and can help us to simplify the design.  
+Using **final** keyword leads programmers to make a round of additional reasoning, to pay more attention to the use of the class. 
+For example, during review a diff with the removal of **final** keyword, it could lead to useful comments to find different solutions. 
+
+There are also some points against **final**. Testing becomes more complex as classes with **final** can not be mocked. 
+A solution to this problem is solved in php, using the annotation **@final**, even if it does not have the same validity as a language keyword. However at nowadays with Ide inspectors, and with static analysis tools like PhpStan (https://phpstan.org/blog), Psalm(https://psalm.dev/docs/) the annotation is fully supporteds and work properly.
+
+A point to disadvantage is that it is considered an instrument too coarse to be effective. 
+A developer can simply to remove the **final** keyword and then do wrong things!. 
+I not agree with this. Too many times I saw to use wrong the inheritance, so I think that this is useful for beginner but also a more senior developer.
+
+To support the last topics I add some links on discussions against the use of **final**.
+
+- [Michael Feathers: It's time to deprecate final](http://butunclebob.com/ArticleS.MichaelFeathers.ItsTimeToDeprecateFinal)
+- [Please stop using "final" classes](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/dddinphp/r9kZ5eI6eiw/TpopVjVv-VgJ)
+
+
+Other links that I used as a starting point for this article
+
+- https://matthiasnoback.nl/2018/09/final-classes-by-default-why/
+- http://ocramius.github.io/blog/when-to-declare-classes-final/
+- https://speakerdeck.com/stuartherbert/the-final-word-about-final?slide=108
+- https://www.reddit.com/r/PHP/comments/2rk3en/when_to_declare_classes_final/
+- https://stackoverflow.com/questions/526312/when-should-i-use-final
+- https://softwareengineering.stackexchange.com/questions/89073/why-is-using-final-on-a-class-really-so-bad
+- https://matthiasnoback.nl/2018/08/negative-architecture-and-assumptions-about-code/
+- http://www.freeklijten.nl/2016/06/17/Final-private-a-reaction
+
+# Conclusion
+
+The theme is quite complex, this article wants to be a general overview. I hope it is useful as a starting point to deepen and begin to evaluate when to use **final** in our classes.
+
+From my point of view the use of keyword **final** can help to improve design. 
+I would not write all my class with **final** but i'd start from the simplest cases: value objects, algorithms, patterns like the template method etc.  
+In any case, with the right attentions it can always be removed, if nothing else it has forcibly added one more step to the analysis process.
```

Comment:
What do you mean by "if nothing else it has forcibly added one more step to the analysis process."? 
Do you mean "se non altro ha aggiunto un ulteriore step al processo di analisi" ?

I would need to understand it better to review it

Reply:
Esatto traduzione corretta in italiano rispetto a quanto volevo scrivere

Reply:
In any case, the **final** keyword can be removed with with due precautions and, if nothing else, using it has forcibly added one more step to the analysis process."